### PR TITLE
Port MStable Strategy to Polygon

### PR DESCRIPTION
**Pull request by idlehusbandry**
# Description

Migrate mstable strategy to polygon.

### CODE REVIEW

File: contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+    /// @notice Claim governance tokens
+    /// @param minTokenToClaim min token to be claimed
+    function claimGovernanceTokens(uint256 minTokenToClaim) external onlyOwner {
+        _claimGovernanceTokens(minTokenToClaim);
+    }
+
+    /// @notice Claim governance tokens
+    /// @param minTokenToClaim min token to be claimed
+    function _claimGovernanceTokens(uint256 minTokenToClaim) internal returns (uint256) {
```
Comment (bugduino) [inline contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol:287]: Can you give me more info on why the changes on `_claimGovernanceTokens` are needed?

File: contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        rewards[0] = _swapGovTokenOnUniswapAndDepositToVault(0); // will redeem whatever possible reward is available
+    }
+
+    /// @notice redeem the rewards. Claims reward as per the _extraData
+    /// @param _extraData must contain the minimum liquidity to receive, and minTokenToClaim
+    /// @return rewards amount of underlyings (mUSD) received after selling rewards
+    function redeemRewards(bytes calldata _extraData) external override onlyIdleCDO returns (uint256[] memory rewards) {
+        (uint256 minLiquidityTokenToReceive, uint256 minTokenToClaim) = abi.decode(_extraData, (uint256, uint256));
+        rewards = new uint256[](2);
+        rewards[1] = _claimGovernanceTokens(minTokenToClaim);
```
[the remark this answers is not in the document]
Comment (idlehusbandry) [reply contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol:141]: I guess there are no round in the polygon.
These are the contracts [Proxy](https://polygonscan.com/address/0x32aBa856Dc5fFd5A56Bcd182b13380e5C855aa29#readProxyContract)
[Implementation](https://polygonscan.com/address/0xfaB26c4a4aE694877E94127fa09c42f0677607E8)

File: contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    function getApr() external view override returns (uint256 apr) {
+        return lastApr;
+    }
+
+    /* -------- internal functions ------------- */
+
+    /// @notice update last saved apr
+    /// @param _amount amount of underlying tokens to mint/redeem
+    function _updateApr(int256 _amount) internal {
+        uint256 mUSDStaked = imUSD.creditsToUnderlying(vault.balanceOf(address(this)));
```
[the remark this answers is not in the document]
Comment (idlehusbandry) [reply contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol:224]: The deposit in vault are properly tokenised in polygon version.
[v-imusd token](https://polygonscan.com/address/0x32aBa856Dc5fFd5A56Bcd182b13380e5C855aa29#readProxyContract)

File: contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol (hunk 1 above)
Comment (idlehusbandry) [reply contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol:287]: function signatures are different ethereum and polygon

Comment (bugduino) [reply contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol:287]: ok but I think we can then simply remove `minTokenToClaim` from here and remove the check `if (rewards + platformRewards > minTokenToClaim)`.
Also do they distribute both MTA and wMATIC? Do you know why there are both `rewards` and `platformRewards`

File: contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (bugduino) [reply contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol:141]: ok let's remove the `minTokenToClaim` then, we will check this off chain eventually but it's not a real issue on polygon

File: contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol (hunk 1 above)
Comment (idlehusbandry) [reply contracts/polygon/strategies/mstable/IdleMStableStrategyPolygon.sol]: There docs say, but I was not able to find the wMATIC rewards. 
I staked my DAI to confirm that. 


I am looking into it.

File: hardhat.config.js [hunk 4]
```diff
@@ -65,27 +65,27 @@ module.exports = {
...
-      //   // blockNumber: 14184625 //  deploy tranche battle winners + mstable completed
-      //   // blockNumber: 14204103 //  deploy tranche battle winners + mstable completed
-      //   // blockNumber: 14217710 //  upgrade mstable tranche
-      //   // blockNumber: 14141000 // harvest strategy
-      //   // blockNumber: 14345651 // upgrade strat with missing setReleaseBlock
-      //   blockNumber: 14366435 // deploy stakingRewards for musd3crv
-      // }
+      forking: {
+        // url: `https://eth-mainnet.alchemyapi.io/v2/${process.env.ALCHEMY_API_KEY}`,
+        url:`https://polygon-mainnet.g.alchemy.com/v2/${process.env.ALCHEMY_API_KEY}`,
```
Comment (bugduino) [inline hardhat.config.js:70]: please comment out this line
